Link to wiki from Readme.

navyasric opened the pull request:
Remove duplication. Moved removed content into wiki.

navyasric commented:
@abhidnya13 Can you please add the badges for CI build, Reference docs. Thanks!

abhidnya13 commented:
I have added the badges. The Travis CI build badge is of dev branch. I am not sure if it should be of dev or master. Let me know and I'll make those changes if needed.

rayluo commented on README.md:
```diff
@@ -1,72 +1,41 @@
 # Microsoft Azure Active Directory Authentication Library (ADAL) for Python
+=====================================
```
The markdown format here isn't exactly right.

A generic tips: you can visualize the markdown locally with some plugins in your browser. Or if you already commit and push, you can [visualize it right from github](https://github.com/AzureAD/azure-activedirectory-library-for-python/blob/nc-readme-updates/README.md).

navyasric replied:
Thanks for catching that.

rayluo commented on README.md:
```diff
@@ -1,72 +1,41 @@
 
-  Upgrade to the latest pip (8.1.2 as of June 2016) and just do `pip install adal`.
+## Versions
+Current version - 0.6.0
```
I would suggest to not mention "current version" here, especially when we are already going to provide the link to the Release history page below. Customers can do a one-click and know the single-source-of-truth about the latest (and, all) version information.

The downside of mentioning it here is it introduces one more place for us to update during future releases. If we could avoid that, then we probably should. :-)

navyasric replied:
I was trying to match other SDKs but I agree with your suggestion of avoiding the duplication.

rayluo commented on README.md:
```diff
@@ -1,72 +1,41 @@
+Current version - 0.6.0
 
-* For Linux
+Minimum recommended version - 0.6.0
```
Again, we probably won't need to hardcode yet another version number here.

On a side note, the idea of "minimum recommended version" is intriguing. What do we want to convey here?

* If the intention is simply want to recommend customers to "always use whatever latest version is available" (which makes perfect sense), we can just write something like that, without hardcoding a specific version number.
* By definition of [Semantic Versioning](https://semver.org/), customers who using an old version X.Y.Z can always safely upgrade to newer version "X.Y-latest.Z-latest", without worrying about breaking change.
* Or customers can technically stay with a version "X.Y.Z", [as long as it still works](https://en.wiktionary.org/wiki/if_it_ain%27t_broke,_don%27t_fix_it).

navyasric replied:
Yes I think specifying that we follow semantic versioning and pointing to the Release page is sufficient.

rayluo replied:
Nitpicking: since we explicitly mention Semantic Versioning in our README, how about we add a link to its official website too?

rayluo commented on README.md:
```diff
@@ -1,72 +1,41 @@
+|[Getting Started](https://github.com/AzureAD/azure-activedirectory-library-for-python/wiki)| [Docs](https://aka.ms/aaddev)| [Samples](https://github.com/azure-samples?query=active-directory)| [Support](README.md#community-help-and-support)
+| --- | --- | --- | --- |
 
-### Install
```
I understand the motivation of "Remove duplication. Moved removed content into wiki."
But do we consider keeping some most-seeking contents, such as "how to install", here in this de-facto project home page? In fact, we [fine tuned](https://github.com/AzureAD/azure-activedirectory-library-for-python/commits/nc-readme-updates/README.md) those installation instructions a couple times before, based on real-world supporting request. Better not lose those valuable knowledge and efforts.

navyasric replied:
We are not losing the instructions recorded over time. They are just in the Wiki page now. I will link to it from the Readme here since it is a "most-seeked" content.

rayluo replied:
Oh I see. Last time I checked, I tried find-by-eye and also "CTRL+F" (i.e. find in page) but could not find keyword "install" in our [Wiki home page](https://github.com/AzureAD/azure-activedirectory-library-for-python/wiki), its sidebar, or the [FAQ page](https://github.com/AzureAD/azure-activedirectory-library-for-python/wiki/FAQ). I was not aware that "install" is inside the ["Basic"](https://github.com/AzureAD/azure-activedirectory-library-for-python/wiki/ADAL-basics).

Now you have kept this "most-seeked" content by a link to that very wiki page. That is a reasonable solution! 👍

rayluo replied:
Oh I see. Last time I checked, I tried find-by-eye and also "CTRL+F" (i.e. find in page) but could not find keyword "install" in our [Wiki home page](https://github.com/AzureAD/azure-activedirectory-library-for-python/wiki), its sidebar, or the [FAQ page](https://github.com/AzureAD/azure-activedirectory-library-for-python/wiki/FAQ). I was not aware that "install" is inside the ["Basic"](https://github.com/AzureAD/azure-activedirectory-library-for-python/wiki/ADAL-basics).

Now you have kept this "most-seeked" content by a link to that very wiki page. That is a reasonable solution! 👍 

Closed.

rayluo commented on README.md:
```diff
@@ -1,72 +1,41 @@
 
-  You'll need a C compiler, libffi + its development headers, and openssl + its development headers.
-  Refer to [cryptography installation](https://cryptography.io/en/latest/installation/)
+> Note: Changes on 'client_id' and 'resource' arguments after 0.1.0
```
Historically, these part showed up early in this document; and then in [one commit](https://github.com/AzureAD/azure-activedirectory-library-for-python/commit/6aaa7c718fdf46c89cd7cc848c9915b1edefed67#diff-04c6e90faac2675aa89e2176d2eec7d8) we purposely move it to the bottom of this document, and the reasoning was documented in that commit.

navyasric replied:
@rayluo Thanks for the context. I placed it under versions since it is information related to a specific version and it is a grey text note to draw little attention. But if this is mostly historical content, is it still relevant and should we remove it instead?

rayluo replied:
@navyasric You raised a very good point! At a hindsight, those paragraph "Changes ... after 0.1.0" really should belong to its release note. I've just updated the [release note on version 0.3.0](https://github.com/AzureAD/azure-activedirectory-library-for-python/releases/tag/0.3.0), you can go ahead to delete this paragraph from README now. Thank you!

rayluo commented on README.md:
```diff
@@ -1,72 +1,41 @@
-  If you still like to build from source, refer to [cryptography installation](https://cryptography.io/en/latest/installation/).
-  For more context, start with this [stackoverflow thread](http://stackoverflow.com/questions/22073516/failed-to-install-python-cryptography-package-with-pip-and-setup-py).
-
-### Acquire Token with Client Credentials
```
Although we do have and mention the "[sample applications on GitHub](https://github.com/azure-samples?utf8=%E2%9C%93&q=active-directory&type=&language=)" later in this README, but those samples locating in a separated repo may or may not cover all the scenarios ADAL Python supports (as of today, they don't). It would seem still beneficial to keep this minimal "out-of-box" samples sections here. What do you think?

navyasric replied:
This content was moved to this  [Wiki page](https://github.com/AzureAD/azure-activedirectory-library-for-python/wiki/Acquire-tokens#adal-python-apis-for-corresponding-flows) which links to relavant dev sample for each flow. I will link to this page from Readme.

rayluo replied:
The other 2 issues in this round of review are really minor, but this one is somewhat different, and needs more discussion.

Historically we had a set of real code snippet right here in the README, later we changed it into the current form as a link to those real samples, after we were called out that they became out-of-sync. See full details [here](https://github.com/AzureAD/azure-activedirectory-library-for-python/pull/103).

Now, moving these content to wiki page by actually duplicating those code snippets there, would bring us back to the potential 2-copies-would-become-out-of-sync situation. In fact, as a real-world example, one of our [current wiki page](https://github.com/AzureAD/azure-activedirectory-library-for-python/wiki/ADAL-basics#usage) technically already outdated, it includes an `api_version=None` which is no longer needed since our latest 1.0.0 release.

So my suggestion is, [anything that can go wrong will go wrong](https://en.wikipedia.org/wiki/Murphy%27s_law), so a better approach is to not create duplicate which we would forget to update later.

rayluo commented on README.md:
```diff
@@ -1,72 +1,41 @@
-### Acquire Token with authorization code
-Find the `Main logic` part in the [sample](sample/website_sample.py#L107-L115) for a complete bare bones web site that makes use of the code below.
-
-## Logging
```
I like that we remove this part, because we do have [a similar page in wiki](https://github.com/AzureAD/azure-activedirectory-library-for-python/wiki/Logging) now.

rayluo commented on README.md:
````diff
@@ -80,29 +49,8 @@ If you find a security issue with our libraries or services please report it to
-``` $ pip install adal ```
-
-### http tracing/proxy
-If you need to bypass self-signed certificates, turn on the environment variable of `ADAL_PYTHON_SSL_NO_VERIFY`
````
AFAIK, this particular line was the only place we document this behavior. If we are going to remove it here, do we plan to document it somewhere in the wiki then?

PS: We did mention it recently in [API reference doc](http://adal-python.readthedocs.io/en/latest/#adal.AuthenticationContext.__init__). But I assume DevOps will not necessarily read API Reference doc. I could be wrong, though.

navyasric replied:
This content is moved to the [Basics page](https://github.com/AzureAD/azure-activedirectory-library-for-python/wiki/ADAL-basics#http-tracingproxy) of the Wiki.

navyasric commented:
@rayluo I have made some updates based on your feedback. Please take a look and let me know. Thanks.

navyasric commented:
@rayluo I have incorporated your feedback to keep the wiki mostly conceptual. I have modified the [acquire token](https://github.com/AzureAD/azure-activedirectory-library-for-python/wiki/Acquire-tokens) section of the wiki to match the format where we link to the code snippet in samples and minimize duplication. (Where there was no sample, I link to method signature). 
This page is linked from the Readme under samples as well as the Basics page in wiki so that we have one copy of this document. Please let me know if this PR is good to merge now.
